cpp/lib/include/ZoteroHarvesterUtil.h

```diff
@@ -498,33 +505,56 @@ template <typename Parameter, typename Result> Result &Future<Parameter, Result>
         } else
             status_ = Status::NO_RESULT;
     }
+}
+
+
+template <typename Parameter, typename Result> Result &Future<Parameter, Result>::getResult() {
+    await();
+    if (status_ != Status::HAS_RESULT)
+        LOG_ERROR("no result for future bound to tasklet " + source_tasklet_->toString());
 
     return *result_;
 }
 
 
 // Tracks harvested records that have been uploaded to the BSZ server.
 class UploadTracker {
-    std::unique_ptr<DbConnection> db_connection_;
 public:
     struct Entry {
         std::string url_;
         std::string journal_name_;
         time_t delivered_at_;
+        std::string delivered_at_str_;
         std::string hash_;
+
+        std::string toString() const;
```

Comment:
Please add, an admittedly unnecessary, "public" access specifier just above this line.